lib/rubygems/printable_uri.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require_relative 'uri_parser'
+
+class Gem::PrintableUri
+  def self.parse_uri(uri)
+    printable_uri = new(uri)
+    printable_uri.parse_uri
+
+    printable_uri
+  end
+
+  def initialize(original_uri)
+    @original_uri = original_uri
+  end
+
+  def parse_uri
+    @original_uri = Gem::UriParser.parse_uri(@original_uri)
+    @uri = @original_uri.dup
+    redact_credential if valid_uri?
+  end
+
+  def valid_uri?
+    @uri.respond_to?(:user) &&
+      @uri.respond_to?(:user=) &&
+      @uri.respond_to?(:password) &&
+      @uri.respond_to?(:password=)
+  end
+
+  def original_password
+    @original_uri.password
+  end
+
+  def to_s
+    @uri.to_s
+  end
+
+  private
+
+  def redact_credential
```

Comment:
This is just some small naming comment, not sure about the grammar behind it, but I've always used `credentials` as plural, even for a single `<user>:<pass>` pair. Using `credential` feels a bit weird to me? Is it just me? Should we use `credential` everywhere?

Reply:
After merging classes into `Gem::Uri` at https://github.com/rubygems/rubygems/pull/4868/commits/9581c2740a37ef12ac0351cb9f166e3d6aa448c1, this method is no longer used so no longer a concern.